File: sdks/typescript/tests/client.test.ts
// file: sdks/typescript/tests/client.test.ts
// version: 1.1.0
// guid: ba3153bc-572d-4084-81ac-63c194ea95f9

import { Client } from '../client/client';
import { ExampleModel } from '../models/model';
import { ExampleService } from '../services/service';

test('callUnary returns response', async () => {
  const client = new Client();
  await client.connect();
  const svc = new ExampleService(client);
  const model = new ExampleModel('1', 'demo');
  const resp = await svc.call(model);
  expect(resp).toHaveProperty('status', 'ok');
  await client.close();
});

test('callUnary requires connection', async () => {
  const client = new Client();
  await expect(client.callUnary('x', {})).rejects.toThrow();
});

File: sdks/java/tests/ClientTest.java
// file: sdks/java/tests/ClientTest.java
// version: 1.0.0
// guid: 6f9d2fdf-152a-41ca-8f0b-df9bd746fbe0

package sdks.java.tests;

import org.junit.jupiter.api.Test;
import sdks.java.client.Client;

/**
 * Tests for Client.
 */
public class ClientTest {
  @Test
  public void testStub() {
    Client client = new Client();
    client.connect();
    // TODO: invoke client methods
    client.close();
  }

  // TODO: add more tests
  // TODO: cover error cases
  // TODO: include integration tests
}
